robot/impl/robot_options.go

```diff
@@ -18,6 +18,8 @@ type options struct {
 
 	// shutdownCallback provides a callback for the robot to be able to shut itself down.
 	shutdownCallback func()
+
+	enableFTDC bool
```

Comment:
I created a `./viam-server -ftdc` flag for this and threaded it through. As mentioned at the prior meeting, can talk with the team if we we want to softly gate this to start.

Reply:
Feels like a fine way to gate (cc @viamrobotics/netcode if others have strong opinions).

Reply:
@dgottlieb sorry for the late comment. this gating approach seems fine to me. 

my one question is, do we anticipate that "ftdc = true" will one day be the default? if so, I think it would be bit more ergonomic to transition this to a "noftdc" flag or something like that with a default of false vs having this existing flag default to true. i think that would play more nicely with how golang sets implicit defaults.

anyway, just a thought - no request to change anything!